packages/backend-defaults/src/entrypoints/database/connectors/postgres.ts

```diff
@@ -52,7 +52,7 @@ export function createPgDatabaseClient(
     database.client.pool.on(
       'createSuccess',
       async (_event: number, pgClient: Client) => {
-        await pgClient.query(`SET ROLE ${role}`);
+        await pgClient.query('SET ROLE $1', [role]);
```

Comment:
Should this be `?` instead? Example of how knex does binding: https://knexjs.org/guide/raw.html#raw-parameter-binding

Reply:
in this case it's the pg client which [their documentation](https://node-postgres.com/features/queries#parameterized-query) says it should work like this. Maybe it's better to have a knex query here? 

I must say I only looked into the context of this file a little bit; the file is called postgres.ts so universal query might not be needed.

Reply:
Ah! Alright maybe that's OK then. Could perhaps have made a `knex.raw` instead but it is what it is.

Only ask from me then, is whether you could double check that there's a test in place that really exercises this, toward the test databases framework (which ensures that it gets run against real dockerized databases)?

Reply:
will do

Reply:
Gentle ping! 🙏

Reply:
I did not find such a test :( this code seems postgres specific. I tried to find an example on how to make a test for it but got a bit lost and then forgot about it 🙈

Reply:
Alright, I tested it by hand now (just adding a `backend.database.role` set to for example `pg_read_all_data` or whatnot. And the backend explodes on startup with `error: syntax error at or near "$1"`. So either way, this isn't the right syntax. Didn't work with `?` or `??` either.

Reply:
https://node-postgres.com/features/queries#parameterized-query

> PostgreSQL does not support parameters for identifiers. If you need to have dynamic database, schema, table, or column names (e.g. in DDL statements) use [pg-format](https://www.npmjs.com/package/pg-format) package for handling escaping these values to ensure you do not have SQL injection!

So yeah, this actually cannot be fixed without explicit escaping - unless we can switch to using the knex client instead. But we do want to do it on the connection level. So ... not sure.

Reply:
that works testing wise 👍 I can dive in a bit more and figure out if there is another way

Reply:
hmm 🤔 what would be wise to do for this? Is it worth to invest now to change to knex?

Reply:
Not sure. There's a database variable just a few lines higher up so at that level it's already present. But the thing is, the knex database object is just a wrapper around an entire connection pool. I believe that the set role command needs to happen at the connection level, so it may be the case that we're stuck with needing to use that exact connection that comes out of the event. So _maybe_ we can't leverage knex there. At least I don't know of a way off the top of my head.

If you can bring in that pg-escape package and perform the escaping manually before putting it into the string, then maybe that also silences your check if it's smart enough.